FB-8 Code display widget

lucas-goldner opened the pull request:
Closes #8 

Now we have an excellent code display widget that does not care about code formatting.

Widget:
![image](https://github.com/lucas-goldner/fluttershow_base/assets/65627237/7d367490-348d-4352-baec-077535bfb2e4)

Code as pasted into the widget:
```dart
// Comment
     class _DartCodeViewerPage extends StatelessWidget {
const _DartCodeViewerPage(this.code);
          final InlineSpan code;

    @override
          Widget build(BuildContext context) {
       final _richTextCode = code;
final _plainTextCode = _richTextCode.toPlainText();
    
        void _showSnackBarOnCopySuccess(dynamic result) {
       ScaffoldMessenger.of(context).showSnackBar(     const SnackBar( content: Text('Copied to Clipboard'),
      ),
           );
  }
    
       void _showSnackBarOnCopyFailure(Object exception) {ScaffoldMessenger.of(context).showSnackBar( SnackBar(
            
            
            
              content: Text('Failure to copy to clipboard: exception'),
  ),
          );
  }
```

lucas-goldner committed:
feat: Add code display widget with two themes

lucas-goldner committed:
feat: Add more code themes

lucas-goldner committed:
feat: Add more code themes

lucas-goldner committed:
feat: Add all other themes

lucas-goldner committed:
feat: Implement priority for coloring

lucas-goldner committed:
test: Add tests for code display widget

lucas-goldner committed:
chore: Replace dart code viewer2 with dart code viewerx

lucas-goldner committed:
fix: Adjust typo

lucas-goldner committed:
feat: Port all themes from dart code viewer to code display themes

immadisairaj commented on example/lib/main.dart:
```diff
@@ -9,13 +13,11 @@ void main() {
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
+    return CupertinoApp(
```
Why `CupertinoApp`? Does it not work with `MaterialApp`?

lucas-goldner replied:
I think those were some leftovers from testing, changed it back though.

immadisairaj commented on lib/components/widgets/wrapper/code_display.dart:
```diff
@@ -0,0 +1,141 @@
+  final String code;
+
+  /// The formatted code.
+  final String formatedCode;
```
I am assuming this can be a private variable as it isn't being used anywhere outside. If this is used for testing, you could add an annotation `@visibleForTesting`.

lucas-goldner replied:
Changed to private 👍🏽

immadisairaj commented on lib/components/widgets/wrapper/code_display.dart:
```diff
@@ -0,0 +1,141 @@
+          formatedCode,
+          width: width,
+          height: height,
+          showCopyButton: false,
```
This also could be an option to users to customise by setting the default to `false`. That way, if they want to show slides + code demo, they could just copy it from here if needed.

Not sure if there would be any limitations.

lucas-goldner replied:
Ok, now that you mention it, I will implement it. The issue here was it only works with MaterialApp since the copy button triggers a ScaffoldMessenger to display a "copied" message. I don't think we need that, so I will update `dart_code_viewer_x` so we can decide wether it should trigger or not.

immadisairaj commented on test/widgets/wrapper/code_display_test.dart:
```diff
@@ -0,0 +1,143 @@
+    final container = tester.widget<Container>(containerFinder);
+    expect(
+      container.decoration,
+      BoxDecoration(
```
Instead of writing this twice, you could create a final variable in the beginning (before pumping the widget) and use the same here. This way, it would be easier.

Similar applies to all the duplicate code here.

lucas-goldner replied:
You are right, refactored a bit. Hope its like you imagined it now.

lucas-goldner committed:
feat: Adjust to requested changes in pr

lucas-goldner committed:
feat: Implements new changes from dart code viewer x into fluttershow base

lucas-goldner commented:
I added now the copy button back. If you dislike the default, you can hide or show it and provide a custom copy button widget. In the example, I changed it back to a Cupertino app and will keep it like that since FlutterShow is also a Cupertino app by default and to preserve testing the copy on non-material apps. It shouldn't matter, tbh.

![image](https://github.com/lucas-goldner/fluttershow_base/assets/65627237/5ac85dc5-c095-48ca-a3f7-02934cee4f26)

lucas-goldner committed:
refactor: Move common finder out to main test group

lucas-goldner committed:
chore: Merge from current release and fix brightness wrapper

lucas-goldner commented:
Also, another issue that I just noticed while integrating the brightness wrapper into this branch is that when using a cupertinoapp the Theme.of does not work as expected, but I forgot about it. So I also fixed it so if either of both is set to darkmode it will trigger the darkThemeChild.
